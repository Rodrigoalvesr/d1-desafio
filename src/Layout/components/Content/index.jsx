import React from 'react';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles({
  root: {
    width: '100%',
  },
});

function Content({ children }) {
  const classes = useStyles();

  return <div className={classes.root}>{children}</div>;
}

export default Content;
