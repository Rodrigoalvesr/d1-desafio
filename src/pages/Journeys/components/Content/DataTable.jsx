import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Skeleton from '@material-ui/lab/Skeleton';
import Status from '@Components/Status';

const useStyles = makeStyles({
  table: {
    width: 900,
  },
  th: {
    fontWeight: 'bold',
    width: '40%',
  },
});

function DataTable({ data, isLoading }) {
  const classes = useStyles();

  return (
    <TableContainer>
      <Table className={classes.table}>
        <TableHead>
          <TableRow>
            <TableCell>Nome</TableCell>
            <TableCell>Destinat√°rio</TableCell>
            <TableCell>Sucesso</TableCell>
            <TableCell>Status</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {isLoading
            ? new Array(4).fill(true).map((item, index) => (
                <TableRow key={index}>
                  <TableCell component="th" scope="row" className={classes.th}>
                    <Skeleton width={120} />
                  </TableCell>
                  <TableCell>
                    <Skeleton width={70} />
                  </TableCell>
                  <TableCell>
                    <Skeleton width={30} />
                  </TableCell>
                  <TableCell>
                    <Skeleton width={80} />
                  </TableCell>
                </TableRow>
              ))
            : data.map((item) => (
                <TableRow key={item.id} className={classes.row}>
                  <TableCell component="th" scope="row" className={classes.th}>
                    {item.name}
                  </TableCell>
                  <TableCell>{item.recipients}</TableCell>
                  <TableCell>{item.success}</TableCell>
                  <TableCell>
                    <Status id={item.status} />
                  </TableCell>
                </TableRow>
              ))}
        </TableBody>
      </Table>
    </TableContainer>
  );
}

export default DataTable;
