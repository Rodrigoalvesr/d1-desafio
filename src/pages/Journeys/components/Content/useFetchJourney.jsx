import { useQuery } from 'react-query';
import api from '@/services/api';

function useFetchJourney(filter) {
  return useQuery(['journey', filter], () =>
    api.get(`/journey/${filter || ''}`).then(({ data }) => data)
  );
}

export default useFetchJourney;
