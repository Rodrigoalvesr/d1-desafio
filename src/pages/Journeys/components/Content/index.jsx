import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import DataTable from './DataTable';
import { useQuery } from 'react-query';
import { useSelector } from 'react-redux';
import api from '@/services/api';

const useStyles = makeStyles({
  root: {
    padding: 54,
  },
});

function Content() {
  const classes = useStyles();
  const { filter } = useSelector((journey) => journey);

  const { isLoading, data = [] } = useQuery(['journey', filter], () =>
    api.get(`/journey/${filter || ''}`).then(({ data }) => data)
  );

  return (
    <div className={classes.root}>
      <DataTable data={data} isLoading={isLoading} />
    </div>
  );
}

export default Content;
