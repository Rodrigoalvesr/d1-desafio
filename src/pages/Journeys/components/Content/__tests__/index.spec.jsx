import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Content from '../index';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import useFetchJourney from '../useFetchJourney';

const mockStore = configureMockStore([thunk]);
const store = mockStore({
  filter: 1,
});
jest.mock('../useFetchJourney', () => jest.fn());

describe('Content', () => {
  it('Renders correctly', async () => {
    useFetchJourney.mockResolvedValue({
      data: [],
    });

    const component = renderer
      .create(
        <Provider store={store}>
          <Content />
        </Provider>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
