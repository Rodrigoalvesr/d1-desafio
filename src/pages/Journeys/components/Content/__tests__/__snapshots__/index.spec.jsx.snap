// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content Renders correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="MuiTableContainer-root"
  >
    <table
      className="MuiTable-root makeStyles-table-2"
      role={null}
    >
      <thead
        className="MuiTableHead-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root MuiTableRow-head"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Nome
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Destinat√°rio
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Sucesso
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Status
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root"
        role={null}
      />
    </table>
  </div>
</div>
`;
